File: ripbull-chat-sdk/src/main/java/com/ripbull/coresdk/utils/CrashReporter.java
package com.ripbull.coresdk.utils;

import io.reactivex.disposables.Disposable;
import java.lang.ref.WeakReference;

public class CrashReporter {

  private WeakReference<DisposableList> list;

  CrashReporter(DisposableList list) {
    this.list = new WeakReference<>(list);
  }

  public CrashReporter() {}

  public void onSubscribe(Disposable d) {
    if (list != null) {
      list.get().add(d);
    }
  }

  public void onNext(Object o) {}

  public void onError(Throwable e) {
    e.printStackTrace();
  }

  public void onComplete() {}
}

File: InAppChat/src/main/java/io/inappchat/inappchat/utils/CrashReporter.kt
package io.inappchat.inappchat.utils

import io.reactivex.rxjava3.core.Observer
import io.reactivex.rxjava3.disposables.Disposable
import java.lang.ref.WeakReference

class CrashReporter<T : Any> : Observer<T> {
    private var list: WeakReference<DisposableList>? = null

    internal constructor(list: DisposableList) {
        this.list = WeakReference(list)
    }

    constructor()

    override fun onSubscribe(d: Disposable) {
        if (list != null) {
            list!!.get()!!.add(d)
        }
    }

    override fun onNext(t: T) {
    }

    override fun onError(e: Throwable) {
        e.printStackTrace()
    }

    override fun onComplete() {}
}